src/ogg_opus.c

```diff
@@ -1043,6 +1046,17 @@ ogg_opus_read_refill (SF_PRIVATE *psf, OGG_PRIVATE *odata, OPUS_PRIVATE *oopus)
 			} ;
 		} ;
 
+	if (oopus->len > oopus->buffersize)
```

Comment:
Did this come out from fuzzing? If `oopus->len > oopus->buffersize` here an error has occurred.

Either opus_decode_... decoded more samples than the buffer can hold (which should never happen as it should return OPUS_BUFFER_TOO_SMALL and fail.)

Or, the end-of-stream padding cut off code above was fed bad data. An alternative would be to alter line 1031:
```
oopus->len = SF_MIN (nsamp, SF_MAX (0, (int) (oopus->pg_pos - oopus->pkt_pos) / oopus->sr_factor)) ;
```
That is, bound `len` to [0, nsamp].

Reply:
Or if `oopus->len > nsamp`, bail out with SFE_MALFORMED_FILE;

Reply:
Yeah, that was something found by fuzzing.

Reply:
Humm. It's just, I cannot see how reallocating would help here. The effect would be that instead of decoded samples, whatever uninitialized data malloc() returns is output. I think returning SFE_MALFORMED_FILE is a pretty low-risk change in this situation.

Reply:
ok, will look at that and the other solution. Then decide if it needs to be fuzzed again.